File: HugoServer/src/main/java/com/hugo/chat/domain/user/UserService.java
package com.hugo.chat.domain.user;


import com.hugo.chat.model.user.dto.UserDTO;

import java.util.Collection;
import java.util.UUID;

public interface UserService {
    UserDTO createUser(UserDTO user);

    Collection<UserDTO> getUsers(UUID roomId);

    void deleteInactiveUser();

    UserDTO updateUser(UserDTO user);

    void setUserActive(String id, String roomId);
}

File: hugo-client/src/services/user/UserService.ts
import UserDTO from "./model/UserDTO";

export default interface UserService {

    createUser(name: string): Promise<UserDTO>;

    keepActive(roomId: string, uuid: string): Promise<void>;

    getUsers(roomId: string): Promise<UserDTO[]>;

    changeName(user: UserDTO): Promise<UserDTO>;

    filterUserDTO(data: UserDTO[], selfId: string): UserDTO[]
}